tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp

```diff
@@ -1424,6 +1517,43 @@ Tensor matmul_1d(
     return output_tensors.at(0);
 }
 
+Tensor matmul_dram_sharded(
+    const Tensor& input_tensor_a,
+    const Tensor& input_tensor_b,
+    std::optional<const Tensor> bias,
+    std::optional<MatmulMultiCoreReuseMultiCastDRAMShardedProgramConfig> program_config,
+    const MemoryConfig& mem_config,
+    std::optional<const DataType> output_dtype,
+    std::optional<const DeviceComputeKernelConfig> compute_kernel_config,
+    bool untilize_out) {
+    std::vector<Tensor> output_tensors = {
+        Tensor(operation::get_workers_for_op_output({input_tensor_a, input_tensor_b}, {bias}))};
+    operation::launch_op(
+        [program_config, mem_config, output_dtype, compute_kernel_config, untilize_out](
+            const std::vector<Tensor>& input_tensors,
+            const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+            const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
+            const auto& input_tensor_a = input_tensors.at(0);
+            const auto& input_tensor_b = input_tensors.at(1);
+
+            auto kernel_config_val =
+                init_device_compute_kernel_config(input_tensor_a.device()->arch(), compute_kernel_config);
+            return {operations::primary::matmul(
+                input_tensor_a,
+                input_tensor_b,
+                optional_input_tensors.at(0),
+                program_config.value(),
+                mem_config,
+                output_dtype,
+                kernel_config_val,
+                untilize_out)};
+        },
+        {input_tensor_a, input_tensor_b},
+        output_tensors,
+        {bias});
+    return output_tensors.at(0);
+}
```

Comment:
Can we put this into the primary matmul path?It doesn't seem like it's doing anything new here.